test/e2e/admin_credentials_test.go

```diff
@@ -0,0 +1,70 @@
+//go:build E2Etests
+
+// Copyright 2025 Microsoft Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package e2e
+
+import (
+	"context"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"k8s.io/apimachinery/pkg/util/rand"
+
+	"github.com/Azure/ARO-HCP/test/util/framework"
+	"github.com/Azure/ARO-HCP/test/util/labels"
+)
+
+var _ = Describe("Request Admin Credentials", func() {
+	It("successfully requests admin credentials for existing cluster",
+		labels.RequireNothing, labels.High, labels.Positive,
+		func(ctx context.Context) {
+			clusterName := "admin-creds-cluster-" + rand.String(6)
+			tc := framework.NewTestContext()
+
+			By("creating resource group and cluster for admin credential request")
+			resourceGroup, err := tc.NewResourceGroup(ctx, "admin-credentials-test", "uksouth")
+			Expect(err).NotTo(HaveOccurred())
+
+			By("deploying cluster for admin credential testing")
+			_, err = framework.CreateBicepTemplateAndWait(ctx,
+				tc.GetARMResourcesClientFactoryOrDie(ctx).NewDeploymentsClient(),
+				*resourceGroup.Name,
+				"admin-creds-cluster",
+				framework.Must(TestArtifactsFS.ReadFile("test-artifacts/generated-test-artifacts/demo.json")),
+				map[string]interface{}{
+					"clusterName": clusterName,
+				},
+				45*time.Minute,
+			)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("requesting admin credentials via ARM API")
+			clusterClient := tc.Get20240610ClientFactoryOrDie(ctx).NewHcpOpenShiftClustersClient()
+			adminRESTConfig, err := framework.GetAdminRESTConfigForHCPCluster(
+				ctx,
+				clusterClient,
+				*resourceGroup.Name,
+				clusterName,
+				10*time.Minute,
+			)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(adminRESTConfig).NotTo(BeNil())
+
+			By("verifying admin credentials work with cluster")
+			Expect(framework.VerifyHCPCluster(ctx, adminRESTConfig)).To(Succeed())
+		})
+})
```

Comment:
I think it's a super heavy lift for us to create a cluster here just to get credentials for it. We also duplicate this test below (and rely on _another_ HCP creation) again effectively - because revoking credentials implies creating them. Is there a test we can piggy-back off of for 1 suite for "Create and Revoke Credentials" together to save cores/cycles?